sql/grant_tables/privilege_set_json.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grant_tables
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// privilegeSetMarshaler handles marshaling duties to and from JSON for a PrivilegeSet.
+type privilegeSetMarshaler struct {
+	GlobalStatic []string
+	Databases    []privilegeSetMarshalerDatabase
+}
+
+// privilegeSetMarshalerDatabase handles marshaling duties to and from JSON for a database in a PrivilegeSet.
+type privilegeSetMarshalerDatabase struct {
+	Name       string
+	Privileges []string
+	Tables     []privilegeSetMarshalerTable
+}
+
+// privilegeSetMarshalerTable handles marshaling duties to and from JSON for a table in a PrivilegeSet.
+type privilegeSetMarshalerTable struct {
+	Name       string
+	Privileges []string
+	Columns    []privilegeSetMarshalerColumn
+}
+
+// privilegeSetMarshalerColumn handles marshaling duties to and from JSON for a column in a PrivilegeSet.
+type privilegeSetMarshalerColumn struct {
+	Name       string
+	Privileges []string
+}
+
+var _ json.Marshaler = PrivilegeSet{}
+var _ json.Unmarshaler = (*PrivilegeSet)(nil)
+
+// MarshalJSON implements the interface json.Marshaler.
+func (ps PrivilegeSet) MarshalJSON() ([]byte, error) {
```

Comment:
`MarshallJSON` and `UnmarshallJSON` could both use some refactoring to split out these nested loops into helper functions

Reply:
I'll definitely do this...in a later PR lol